Code refactor

rloterh opened the pull request:
In this branch we implemented:

- Code refactor.
- Clean code.
- methods to improve UI.
- Improve UI readability.

rukundoeric committed:
Add main.rb && Create UI functions

rukundoeric committed:
Fix Linters

rukundoeric committed:
Update Readme

rukundoeric committed:
Fix linters

rloterh committed:
Remove comments

rloterh committed:
Fix linter errors

rloterh committed:
Update ReadMe

rloterh committed:
Add exit_remarks method

rloterh committed:
Add dialogue separators

elisha2kyakpo1 commented on main.rb:
```diff
@@ -0,0 +1,162 @@
+require_relative './teacher'
+require_relative './book'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_books
+    @books.each do |book|
+      puts "Title: \"#{book.title}\", Author: #{book.author}"
+    end
+  end
+
+  def list_people
+    @people.each do |person|
+      puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    person_type = gets.chomp
+    if person_type != '1' && person_type != '2'
+      puts 'Invalid option'
+      return
+    end
+
+    print 'Age: '
```
- [x] the role of main.rb is only to start the code and ask the user for the initial choice of options (or to quit). This code should be in another file, for example, app.rb

elisha2kyakpo1 commented on main.rb:
```diff
@@ -0,0 +1,162 @@
+    print 'Title: '
+    title = gets.chomp
+
+    print 'Author: '
+    author = gets.chomp
+
+    @books << Book.new(title, author)
+    puts 'Book created successfully'
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}"
+    end
+    book_index = gets.chomp.to_i
+
+    puts "\nSelect a person from the following list by number (not id)"
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+
+    person_index = gets.chomp.to_i
+
+    print "\nDate: "
+    date = gets.chomp
+
+    @rentals << Rental.new(date, @books[book_index], @people[person_index])
+    puts 'Rental created successfully'
+  end
```
- [x] the role of main.rb is only to start the code and ask the user for the initial choice of options (or to quit). This code should be in another file, for example, app.rb

elisha2kyakpo1 commented on main.rb:
```diff
@@ -0,0 +1,162 @@
+    print 'Name: '
+    name = gets.chomp
+
+    person =
+      case person_type
+      when '1'
+        print 'Has parent permission? [Y/N]: '
+        parent_permission = gets.chomp
+        parent_permission = parent_permission.downcase == 'y'
+
+        Student.new(age, name, parent_permission)
+      when '2'
+        print 'Specialization: '
+        specialization = gets.chomp
+
+        Teacher.new(age, specialization, name)
+      end
+
+    @people << person
+    puts 'Person created successfully'
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+
```
- [x] the role of main.rb is only to start the code and ask the user for the initial choice of options (or to quit). This code should be in another file, for example, app.rb

elisha2kyakpo1 commented on main.rb:
```diff
@@ -0,0 +1,162 @@
+    puts 'Rental created successfully'
+  end
+
+  def list_rentals_for_person_id
+    print 'ID of person: '
+    id = gets.chomp.to_i
+
+    rentals = @rentals.filter { |rental| rental.person.id == id }
+    puts 'Rentals:'
+    rentals.each do |rental|
+      puts "Date: #{rental.date}, Book \"#{rental.book.title}\" by #{rental.book.author}"
+    end
+  end
+end
```
- [x] the role of main.rb is only to start the code and ask the user for the initial choice of options (or to quit). This code should be in another file, for example, app.rb

elisha2kyakpo1 reviewed (changes requested):
Hi, @rloterh,

Great job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

rloterh committed:
Refactor app class

rloterh committed:
Refactor inputs and outputs

rloterh committed:
Clean syntax

rloterh committed:
Refactor inputs and outputs

rloterh committed:
Refactor operations logic to external class from main

rloterh commented:
**In these changes, I:**
- Refactored classes and methods in the main class into separate classes.
- Refactored inputs and output methods.
- Improve UI.
- Cleaned syntax.
